restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/telephony/proxy/ProxyManagerTest.java

```diff
@@ -20,70 +29,140 @@
+        tool1 = new SipStackTool("ImsClientsDialTest1");
+        tool2 = new SipStackTool("ImsClientsDialTest2");
     }
 
     @Before
     public void before() throws Exception {
-        receiver = tool.initializeSipStack(SipStack.PROTOCOL_UDP, "127.0.0.1", "5070", "127.0.0.1:5080");
-        phone = receiver.createSipPhone("127.0.0.1", SipStack.PROTOCOL_UDP, 5080, "sip:alice@127.0.0.1:5070");
+        
+        imsSipStack = tool1.initializeSipStack(SipStack.PROTOCOL_UDP, "127.0.0.1", "5060", "127.0.0.1:5080");
+
+        augustSipStack = tool2.initializeSipStack(SipStack.PROTOCOL_UDP, "127.0.0.1", "5092", "127.0.0.1:5080");
+        augustPhone = augustSipStack.createSipPhone("127.0.0.1", SipStack.PROTOCOL_UDP, 5080, augustContact);
+        imsAugustPhone = imsSipStack.createSipPhone("127.0.0.1", SipStack.PROTOCOL_UDP, 5080, augustContact);
+        imsAugustPhone.setLoopback(true);
     }
 
+    private String adminAccountSid = "ACae6e420f425248d6a26948c17a9e2acf";
+    private String adminAuthToken = "77f8c12cc7b8f8423e5c38b035249166";
+
     @After
     public void after() throws Exception {
-        if (phone != null) {
-            phone.dispose();
+        if (augustPhone != null) {
+            augustPhone.dispose();
+        }
+        if (augustSipStack != null) {
+            augustSipStack.dispose();
         }
-        if (receiver != null) {
-            receiver.dispose();
+
+        if (imsSipStack != null) {
+            imsSipStack.dispose();
+        }
+        if (imsAugustPhone != null) {
+            imsAugustPhone.dispose();
         }
+        
         deployer.undeploy("ProxyManagerTest");
     }
 
     @Test
-    public void testRestCommRegistration() throws Exception {
+    public void testRegisterWithGateWay() throws ParseException, InterruptedException, SQLException {
         deployer.deploy("ProxyManagerTest");
-        // This is necessary for SipUnit to accept unsolicited requests.
-        phone.setLoopback(true);
-        phone.listenRequestMessage();
-        // Wait for the SIP REGISTER request.
-        final RequestEvent register = phone.waitRequest(75 * 1000);
-        // Validate the request.
-        final String method = register.getRequest().getMethod();
-        assertTrue("REGISTER".equalsIgnoreCase(method));
-        // Send the OK response.
-        final MessageFactory factory = receiver.getMessageFactory();
-        final Request request = register.getRequest();
-        Response ok = factory.createResponse(200, request);
-        SipTransaction transaction = phone.sendReply(register, ok);
-        // Validate the transaction.
-        assertNotNull(transaction);
+        SipURI uri = augustSipStack.getAddressFactory().createSipURI(null, "127.0.0.1:5080");
+        final String userName = "august@127.0.0.1:5092";
+        isAuthorizationHasRightUserName = false;
+        isRegisterRequestReceived = false;
+        ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
any reason to do this async to the test thread, why dont ew simple set the Gateway,and then proceed with traffic assertions?

Reply:
we may need to declare those asserting flags as volatile if several threads are read/writing them..

Reply:
As the time, copied pasted without evaluating. 
I removed the parallel code and also removed the check variables, put direct assert at the point as we don't use thread any more.